divide code into components

NormalDuck opened the pull request:
There are 3 new components:
`ScoringInput`: A wrapper around `NumberInput` but you can add text
`CheckboxText`: Checkbox but you can add text to describe what the checkbox does
`PhaseToggle`: A component that takes a array and builds a button group.
`ScoringSection`: Divides `ScoringInput` into sections

Jxwoon commented on app/auto/page.tsx:
```diff
@@ -21,111 +24,96 @@ export default function Home() {
             onChange={(val) => {
               setLocalDraft({
                 ...localDraft,
-                auto: { ...localDraft.auto, highSpecimen: val },
```
Missing the specimen logic

NormalDuck replied:
fixed it

Jxwoon commented on app/end/page.tsx:
```diff
@@ -17,79 +20,70 @@ export default function Home() {
-            defaultValue={localDraft.end.highNet}
+        <ScoringSection sectionName="Scored Sample">
+          <ScoringInput
+            defaultValue={localDraft.end.net}
```
NIT: fix scheme

NormalDuck commented:
I have made changes to the most of the schema and added some new components. The visuals should match the actual matches. I didn't remove `final` page since I might need it later.

johnylamw reviewed (changes requested):
AscentMap needs to modified to also include a "none" state. Right now, the options are either "Observation" or "Level x". There should be an option for when a robot does neither of these.

Additionally, refactor all instances of "Acent" to "Ascent" for proper spelling since it may confuse people working on the code. There might be a select all option in your code editor to refactor all instances of it rather than needing you to do it by hand.

johnylamw commented:
Also, a fun idea would be to create a VERTICAL slider for ascent states. Higher on the slider means the robot reached a higher climb. But that will require you to add labels to the axis. Maybe add some color coding to designate the level they're at. 

(just a fun idea if you're bored. I think the drop-downs that you have are more intuitive though and simpler to manage imo)

NormalDuck commented:
I have fixed **all** the typos with cspell. Added a new component in the newest commit and refactored the code to use the new component called `SelectInputText` Added none for ascent.

johnylamw reviewed (changes requested):
"Observation" option missing for end-game and there is now 2 options for level 3. 
(small typos like these are natural - that's why pull requests exist)

There's also a "test" SelectInputText on the End pages. General practice would be to comment out or remove any sort of test interfaces considering that you're pushing into main which will be used in production. Personally I don't have any problems with that but just keep that in mind when you start working/interning in the industry. You can keep it there if you plan on further testing with it.

Overall, I like the new changes w/ the components and good job on fixing the spelling mistakes.

NormalDuck commented:
I didn't look carefully when I pushed the code😅

johnylamw reviewed (approved):
Good on my end. 
@Jxwoon thoughts?

Jxwoon commented on app/end/page.tsx:
```diff
@@ -1,164 +1,191 @@
+          <option>Level1</option>
+          <option>Level2</option>
+          <option>Level3</option>
+          <option>Level3</option>
```
Nit: why is there two level3?

NormalDuck replied:
I didn't look carefully my bad

Jxwoon reviewed (approved):
Don't want to hold up this PR. For the most part this looks good to go! Just left one nit comment that could be addressed in another PR or this one
